src/arkode/arkode_erkstep.c

```diff
@@ -847,6 +939,207 @@ int erkStep_TakeStep(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPtr)
+    retval = step_mem->f(ark_mem->tcur, sens_tmp, stage_values[is],
+                         ark_mem->user_data);
+    step_mem->nfe++;
+
+    /* The checkpoint was not found, so we need to recompute at least
+       this step forward in time. We first seek the last checkpointed step
+       solution, then recompute from there. */
+    if (retval > 0)
+    {
+      N_Vector checkpoint = N_VGetSubvector_ManyVector(ark_mem->tempv3, 0);
+      int64_t curr_step, start_step;
+      curr_step = start_step = ark_mem->adj_step_idx;
+
+      SUNErrCode errcode = SUN_ERR_CHECKPOINT_NOT_FOUND;
+      for (int64_t i = 0; i <= curr_step; ++i, --start_step)
+      {
+        SUNDIALS_MAYBE_UNUSED int64_t stop_step = curr_step + 1;
+        SUNLogDebug(ARK_LOGGER, "ARKODE::erkStep_TakeStep_Adjoint",
+                    "searching-for-checkpoint",
+                    "start_step = %li, stop_step = %li", start_step, stop_step);
+        sunrealtype checkpoint_t;
+        errcode =
+          SUNAdjointCheckpointScheme_LoadVector(ark_mem->checkpoint_scheme,
+                                                start_step, step_mem->stages, 1,
+                                                &checkpoint, &checkpoint_t);
+        if (errcode == SUN_SUCCESS)
+        {
+          /* OK, now we have the last checkpoint that stored as (start_step, stages).
+             This represents the last step solution that was checkpointed. As such, we
+             want to recompute from start_step+1 to stop_step. */
+          start_step++;
+          sunrealtype t0 = checkpoint_t;
+          sunrealtype tf = ark_mem->tn;
+          SUNLogDebug(ARK_LOGGER, "ARKODE::erkStep_TakeStep_ERK_Adjoint",
+                      "start-recompute",
+                      "start_step = %li, stop_step = %li, t0 = %" SUN_FORMAT_G
+                      ", tf = %" SUN_FORMAT_G "",
+                      start_step, stop_step, t0, tf);
+          if (SUNAdjointStepper_RecomputeFwd(adj_stepper, start_step, t0, tf,
+                                             checkpoint))
+          {
+            return (ARK_ADJ_RECOMPUTE_FAIL);
+          }
+          SUNLogDebug(ARK_LOGGER, "ARKODE::erkStep_TakeStep_ERK_Adjoint",
+                      "end-recompute",
+                      "start_step = %li, stop_step = %li, t0 = %" SUN_FORMAT_G
+                      ", tf = %" SUN_FORMAT_G "",
+                      start_step, stop_step, t0, tf);
+          return erkStep_TakeStep_Adjoint(ark_mem, dsmPtr, nflagPtr);
+        }
+      }
+      if (errcode != SUN_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+    }
+    else if (retval < 0) { return (ARK_RHSFUNC_FAIL); }
+  }
+
+  /* Throw away the step solution */
+  sunrealtype checkpoint_t = 0.0;
+  N_Vector checkpoint      = N_VGetSubvector_ManyVector(ark_mem->tempv2, 0);
+  SUNErrCode errcode =
+    SUNAdjointCheckpointScheme_LoadVector(ark_mem->checkpoint_scheme,
+                                          ark_mem->adj_step_idx, 0, 0,
```

Comment:
Same suggestion as in ERKStep
```suggestion
                                          ark_mem->adj_step_idx, 0, /*peek=*/SUNFALSE,
```